src/test/java/uk/gov/ida/matchingserviceadapter/services/AttributeStatementBuilder.java

```diff
@@ -0,0 +1,136 @@
+import org.opensaml.saml.saml2.core.AttributeStatement;
+import uk.gov.ida.saml.core.IdaConstants;
+import uk.gov.ida.saml.core.OpenSamlXmlObjectFactory;
+import uk.gov.ida.saml.core.extensions.eidas.BirthName;
+import uk.gov.ida.saml.core.extensions.eidas.CurrentFamilyName;
+import uk.gov.ida.saml.core.extensions.eidas.CurrentGivenName;
+import uk.gov.ida.saml.core.extensions.eidas.DateOfBirth;
+import uk.gov.ida.saml.core.extensions.eidas.Gender;
+import uk.gov.ida.saml.core.extensions.eidas.PersonIdentifier;
+import uk.gov.ida.saml.core.extensions.eidas.PlaceOfBirth;
+import uk.gov.ida.saml.core.extensions.eidas.impl.BirthNameBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.CurrentFamilyNameBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.CurrentGivenNameBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.DateOfBirthBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.GenderBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.PersonIdentifierBuilder;
+import uk.gov.ida.saml.core.extensions.eidas.impl.PlaceOfBirthBuilder;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class AttributeStatementBuilder {
+
+    public static final String BIRTH_NAME="http://eidas.europa.eu/attributes/naturalperson/BirthName";
+    public static final String PLACE_OF_BIRTH="http://eidas.europa.eu/attributes/naturalperson/PlaceOfBirth";
+
+    private static OpenSamlXmlObjectFactory openSamlXmlObjectFactory = new OpenSamlXmlObjectFactory();
+    private List<Attribute> attributes = new ArrayList<>();
+
+    public static AttributeStatementBuilder anAttributeStatement() {
+        return new AttributeStatementBuilder();
+    }
+
+    /**
+     * Retain original defaults to maintain backward compatibility with existing tests.
+     * @deprecated Use @link AttributeStatementBuilder#anEidasAttributeStatement(Attribute ... attributes) instead for future tests.
+     * @return
+     */
+    public static AttributeStatementBuilder anEidasAttributeStatement() {
+        return anEidasAttributeStatement(
+            aCurrentGivenNameAttribute("Joe"),
+            aCurrentFamilyNameAttribute("Bloggs"),
+            aPersonIdentifierAttribute("JB12345"),
+            aDateOfBirthAttribute(LocalDate.now()));
+    }
+
+    public static AttributeStatementBuilder anEidasAttributeStatement(Attribute ... attributes) {
+        return anAttributeStatement().addAllAttributes(attributes != null ? Arrays.asList(attributes) : Collections.emptyList());
+    }
+
+    public static Attribute aCurrentGivenNameAttribute(String givenName) {
+        Attribute firstNameAttr =  anAttribute(IdaConstants.Eidas_Attributes.FirstName.NAME);
+        CurrentGivenName firstNameValue = new CurrentGivenNameBuilder().buildObject();
+        firstNameValue.setFirstName(givenName);
+        firstNameAttr.getAttributeValues().add(firstNameValue);
+        return firstNameAttr;
```

Comment:
Minor point: I realise the CurrentGivenNameBuilder class etc aren't part of this PR but should they be written so that it's possible to do something like this?
`CurrentGivenName firstNameValue =  new CurrentGivenNameBuilder().withFirstName(givenName).build()`

Reply:
Agreed! 👍 
Consistency is key.